libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -82,8 +82,33 @@ internal class RouteAlternativesController constructor(
         }
     }
 
-    fun triggerAlternativeRequest() {
-        nativeRouteAlternativesController.refreshImmediately()
+    fun triggerAlternativeRequest(listener: RouteAlternativesRequestCallback?) {
+        nativeRouteAlternativesController.refreshImmediately { expected ->
+            val routeProgress = tripSession.getRouteProgress()
+                ?: run {
+                    listener?.onRouteAlternativesAborted(
+                        """
+                            |Route progress not available, ignoring alternatives update.
+                            |Continuous alternatives are only available in active guidance.
+                        """.trimMargin()
+                    )
+                    return@refreshImmediately
+                }
```

Comment:
Should we check this before calling `refreshImmediately`?

Reply:
I think we need to do it after - an example could be a case where we request alternatives and while we wait for a response the primary route changes or is cleared.